interview-questions.md

```diff
@@ -7,44 +8,77 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
+Your answer:
 
-  Researched answer:
+- The foreign key would be named Sightings, and it would be part of the Animal model to connect to the has_many Sightings model.
 
+Researched answer:
 
+- The foreign key would be named Animal and would be connect to the has_many Sightings as the foreign key to ID the animals
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+Your answer:
 
-  Researched answer:
+- Create, so hackers cant hack into the website.
 
+Researched answer:
 
+- def create must always be passed params to protect the data being submitted into the database isnt maliscious.
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+Your answer:
 
-  Researched answer:
+1. rails generate new model - allows you to create a new model application
+2. rails generate controller - allows you to create a controller for your application.
+3. rails
 
+Researched answer:
 
+1. rails generate new model - allows you to create a new model application
+2. rails generate controller - allows you to create a controller for your application.
+3. rails generate migration - adds a column to the existing model
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-method="GET"    /users          
+method="GET" /users
+
+- This GET would use the Restful route Index, to display all of the users.
+
+method="GET" /users/1
+
+- This GET would use the Restful route Show, to display only the single user thats at the id of 1.
+
+method="GET" /users/new
+
+- This GET would use the Restful route New, to create a new user instance.
+
+method="GET" /users/edit/1
 
-method="GET"    /users/1  
+- This GET would use the Restful route Edit, to edit the existing user at the id of 1.
 
-method="GET"    /users/new
+method="POST" /users/
 
-method="GET"    /users/edit/1    
+- This POST would use the Restful route Create, to create a new user on our application.
 
-method="POST"   /users/       
+method="PUT" /users/1
 
-method="PUT"    /users/1      
+- This PUT would use the Restful route Update, to update the existing user items at id of 1.
```

Comment:
'user#update'

This makes the change for an update to a post actually change in the database.

Reply:
So, I see what I did wrong here, I put what would be in the browser, not the method to the restful routes. Thank you.